src/repositories/vocabulary.repository.ts

```diff
@@ -36,18 +52,31 @@ class VocabularyRepository {
   async update(vocabulary: IVocabulary): Promise<IVocabulary> {
     const oldVocabulary = await this.getById(vocabulary._id!);
     oldVocabulary.words = vocabulary.words;
-    await vocabulariesDb.insert(oldVocabulary);
-    return oldVocabulary;
+    try {
+      await vocabulariesDb.insert(oldVocabulary);
+      return oldVocabulary;
+    } catch (error) {
+      if ((error as any).statusCode == 404) {
+        throw new HttpException(
+          HttpStatusCode.NOT_FOUND,
+          'vocabulary not found by id! repository',
```

Comment:
Please start all exception messages from capital letter, and why do you write `id! repository`, why this exclamation mark? and for what to write repository?